datastore/client.go

```diff
@@ -44,13 +45,18 @@ type datastoreClient struct {
 
 func newDatastoreClient(conn grpc.ClientConnInterface, projectID, databaseID string) pb.DatastoreClient {
 	resourcePrefixValue := "projects/" + projectID
-	if databaseID != "" {
+	reqParamsHeaderValue := "project_id=" + url.QueryEscape(projectID)
+
+	if databaseID != DefaultDatabaseID {
```

Comment:
I think you can also keep the empty check

Reply:
Done